src/main/java/service/FruitShopService.java

```diff
@@ -0,0 +1,36 @@
+package service;
+
+import exception.OperationException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import model.OperationType;
+import strategy.OperationHandler;
+
+public class FruitShopService {
+    private final Map<String, Integer> fruitInventory = new HashMap<>();
```

Comment:
Follow the encapsulation principle. Consider making 'fruitInventory' private and provide a getter method to access it. Direct access to the map outside this class should be avoided to maintain encapsulation.